### PR TITLE
test: migrate to vitest

### PR DESCRIPTION
**Pull request by danielroe**
This migrates the test suite to `vitest` preparatory to adding some tests for `@pinia/nuxt`, which wasn't playing nicely with jest.

Previous: https://github.com/vuejs/pinia/pull/1365

### CODE REVIEW

File: packages/pinia/__tests__/ssr.spec.ts [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /**
  * @jest-environment node
```
Comment (posva) [inline packages/pinia/__tests__/ssr.spec.ts]: should this be changed too?

File: vite.config.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+        '**/src/index.ts',
+        '**/*.d.ts',
+        '**/src/devtools',
+        '**/src/hmr.ts',
+        '**/src/deprecated.ts',
+        '**/src/vue2-plugin.ts',
+      ],
+    },
+    environment: 'happy-dom',
+    globals: true, // Specifically to make createTestingPinia happy
```
Comment (posva) [inline vite.config.ts:40]: what do you mean? 😮

Comment (danielroe) [reply vite.config.ts:40]: by default vitest doesn't expose global objects as jest does, so `vi` is not defined by default. We could either update the createTestingPinia tests to pass vi.fn. Or we can use this globals option.

File: packages/pinia/__tests__/ssr.spec.ts (hunk 1 above)
Comment (danielroe) [reply packages/pinia/__tests__/ssr.spec.ts]: Good point. Vitest does respect it, but we might as well!

File: vite.config.ts (hunk 2 above)
Comment (posva) [reply vite.config.ts:40]: as long as the type of the variable `createSpy()` isn't `any` in the source code 😆

**Review by posva — APPROVED** [review]

**Pull request merged**